src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -40,61 +40,66 @@ public TestConverter(TestLogger logger, string sourceAssembly)
 
             // Return cached value if we have one
-            if (vsTestCaseMap.ContainsKey(test.TestName.UniqueName))
-                return vsTestCaseMap[test.TestName.UniqueName];
+            string id = testNode.GetAttribute("id");
+            if (vsTestCaseMap.ContainsKey(id))
+                return vsTestCaseMap[id];
            
             // Convert to VS TestCase and cache the result
-            var testCase = MakeTestCaseFromNUnitTest(test);
-            vsTestCaseMap.Add(test.TestName.UniqueName, testCase);
+            var testCase = MakeTestCaseFromXmlNode(testNode);
+            vsTestCaseMap.Add(id, testCase);
             return testCase;             
         }
 
-        public TestCase GetCachedTestCase(string key)
+        public TestCase GetCachedTestCase(string id)
         {
-            if (vsTestCaseMap.ContainsKey(key))
-                return vsTestCaseMap[key];
+            if (vsTestCaseMap.ContainsKey(id))
+                return vsTestCaseMap[id];
 
-            logger.SendErrorMessage("Test " + key + " not found in cache");
+            logger.SendErrorMessage("Test " + id + " not found in cache");
             return null;
         }
 
-        public VSTestResult ConvertTestResult(NUnitTestResult result)
+        public VSTestResult ConvertTestResult(XmlNode resultNode)
         {
-            TestCase ourCase = GetCachedTestCase(result.Test.TestName.UniqueName);
+            TestCase ourCase = GetCachedTestCase(resultNode.GetAttribute("id"));
             if (ourCase == null) return null;
 
             VSTestResult ourResult = new VSTestResult(ourCase)
-                {
-                    DisplayName = ourCase.DisplayName,
-                    Outcome = ResultStateToTestOutcome(result.ResultState),
-                    Duration = TimeSpan.FromSeconds(result.Time)
-                };
+            {
+                DisplayName = ourCase.DisplayName,
+                Outcome = GetTestOutcome(resultNode),
+                Duration = TimeSpan.FromSeconds(resultNode.GetAttribute("duration", 0.0))
+            };
+
+            var startTime = resultNode.GetAttribute("start-time");
+            if (startTime != null)
+                ourResult.StartTime = DateTimeOffset.Parse(startTime);
+
+            var endTime = resultNode.GetAttribute("end-time");
+            if (endTime != null)
+                ourResult.EndTime = DateTimeOffset.Parse(endTime);
 
             // TODO: Remove this when NUnit provides a better duration
             if (ourResult.Duration == TimeSpan.Zero && (ourResult.Outcome == TestOutcome.Passed || ourResult.Outcome == TestOutcome.Failed))
                 ourResult.Duration = TimeSpan.FromTicks(1);
+
             ourResult.ComputerName = Environment.MachineName;
 
-            // TODO: Stuff we don't yet set
-            //   StartTime   - not in NUnit result
-            //   EndTime     - not in NUnit result
```

Comment:
If StartTime and EndTime are in the result now, should we add them to the VS test result? Did you just delete because they still aren't available?